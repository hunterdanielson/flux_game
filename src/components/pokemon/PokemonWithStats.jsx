import React from 'react';
import PropTypes from 'prop-types';
import Pokemon from './Pokemon';
import Stats from '../stats/Stats';

const PokemonWithStats = ({ pokemon }) => (
  <section>
    <Pokemon image={pokemon.image} />
    <Stats health={pokemon.health} attack={pokemon.attack} defense={pokemon.defense} />
  </section>
);

PokemonWithStats.propTypes = {
  pokemon: PropTypes.shape({
    image: PropTypes.string.isRequired,
    health: PropTypes.number.isRequired,
    attack: PropTypes.number.isRequired,
    defense: PropTypes.number.isRequired,
  })
};

export default PokemonWithStats;
