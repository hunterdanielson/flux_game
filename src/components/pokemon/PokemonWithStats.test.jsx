import React from 'react';
import { shallow } from 'enzyme';
import PokemonWithStats from './PokemonWithStats';

describe('Pokemon component', () => {
  it('renders Pokemon', () => {
    const pokemon = {
      image: 'http://someimg.png',
      health: 100,
      attack: 10,
      defense: 10
    };
    const wrapper = shallow(<PokemonWithStats pokemon={pokemon}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
