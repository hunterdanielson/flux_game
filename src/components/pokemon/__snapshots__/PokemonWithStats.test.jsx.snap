// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon component renders Pokemon 1`] = `
<section>
  <Pokemon
    image="http://someimg.png"
  />
  <Stats
    attack={10}
    defense={10}
    health={100}
  />
</section>
`;
